src/Stratis.SmartContracts.CLR/ContractModuleDefinition.cs

```diff
@@ -48,26 +49,51 @@ public TypeDefinition ContractType
         public ModuleDefinition ModuleDefinition { get; private set; }
 
         /// <inheritdoc />
-        public void Rewrite(IILRewriter rewriter)
+        public bool Rewrite(IILRewriter rewriter)
         {
-            this.ModuleDefinition = rewriter.Rewrite(this.ModuleDefinition);
+            try
+            {
+                this.ModuleDefinition = rewriter.Rewrite(this.ModuleDefinition);
+                return true;
+            }
+            catch (Exception)
+            {
+                return false;
```

Comment:
Should log here, and on line 79 below in this file

Reply:
I've moved all the exception handling to `ReflectionVirtualMachine` and added logging.